src/contexts/addresses.tsx

```diff
@@ -32,12 +32,12 @@ import { getHumanReadableError } from '../utils/api'
 import { NetworkType } from '../utils/settings'
 import { useGlobalContext } from './global'
 
-export type TransactionType = 'consolidation' | 'transfer' | 'sweep'
+export type TransactionType = 'consolidation' | 'transfer' | 'sweep' | 'contract'
 
 type SimpleTx = {
   txId: string
   fromAddress: string
-  toAddress: string
+  toAddress?: string
```

Comment:
This type starts to be a bit to generic IMHO :) 

We could maybe define a `BaseTX` type with all required parameters, which could then be extended as `TransferTX`, `ContractTX` with specific params etc. Something like:

```ts
type BaseTx = {
  txId: string
  fromAddress: string
  timestamp: number
  type: TransactionType
  network: NetworkType
}

type TransferTx = BaseTx & { toAddress: string, amount: bigint }
```

WDYT?

Reply:
Also, let's extract Transaction types to their own file (in the `types` folder)?

This would allow us to add some handy type predicates there, like `isContractTx` or `isScriptTx`.

Reply:
Yerp :+1:

Reply:
I did rebase and I didnt see it renamed?.. :s